bsb_neuron/adapter.py

```diff
@@ -170,50 +170,60 @@ def _allocate_transmitters(self, simulation):
         simdata.transmap = self._map_transceivers(simulation, simdata)
 
     def _map_transceivers(self, simulation, simdata):
-        blocks = []
         offset = 0
         transmap = {}
 
-        for cm, cs in simulation.get_connectivity_sets().items():
-            # For each connectivity set, determine how many unique transmitters they will place.
-            pre, _ = cs.load_connections().as_globals().all()
-            all_cm_transmitters = np.unique(pre[:, :2], axis=0)
-            # Now look up which transmitters are on our chunks
-            pre_t, _ = cs.load_connections().from_(simdata.chunks).as_globals().all()
-            our_cm_transmitters = np.unique(pre_t[:, :2], axis=0)
-            # Look up the local ids of those transmitters
-            pre_lc, _ = cs.load_connections().from_(simdata.chunks).all()
-            local_cm_transmitters = np.unique(pre_lc[:, :2], axis=0)
-
-            # Find the common indexes between all the transmitters, and the
-            # transmitters on our chunk.
-            dtype = ", ".join([str(all_cm_transmitters.dtype)] * 2)
-            _, _, idx_tm = np.intersect1d(
-                our_cm_transmitters.view(dtype),
-                all_cm_transmitters.view(dtype),
-                assume_unique=True,
-                return_indices=True,
-            )
+        pre_types = set(cs.pre_type for cs in simulation.get_connectivity_sets().values())
+        for pre_type in pre_types:
+            data = []
+            for cm, cs in simulation.get_connectivity_sets().items():
+                if cs.pre_type != pre_type:
+                    continue
+                pre, _ = cs.load_connections().as_globals().all()
+                data.append(pre[:, :2])
+
+            # Save all transmitters of the same pre_type across connectivity sets
+            all_cm_transmitters = np.unique(data, axis=0)
```

Comment:
Does this work without concatenating the data block?

Reply:
no, right :)